build.cmd

```diff
@@ -38,21 +38,18 @@ echo Commencing CoreCLR Repo build
 echo.
 
 :: Set the remaining variables based upon the determined build configuration
-set "__BinDir=%__RootBinDir%\Product\%__BuildArch%\%__BuildType%"
+set "__BinDir=%__RootBinDir%\Product\%__BuildOS%.%__BuildArch%.%__BuildType%"
+set "__IntermediatesDir=%__RootBinDir%\intermediates\%__BuildOS%.%__BuildArch%.%__BuildType%"
 set "__PackagesBinDir=%__BinDir%\.nuget"
 set "__ToolsDir=%__RootBinDir%\tools"
-set "__TestWorkingDir=%__RootBinDir%\tests\%__BuildArch%\%__BuildType%"
-set "__IntermediatesDir=%__RootBinDir%\intermediates\%__BuildArch%\%__BuildType%"
+set "__TestWorkingDir=%__RootBinDir%\tests\%__BuildOS%.%__BuildArch%.%__BuildType%"
```

Comment:
Can you do the same for tests\buildtest.cmd, though technically the above will override whatever is in tests\buildtest.cmd, we should still be able to build tests independently and it would help maintain consistency